cli/wallet/file/file.go

```diff
@@ -0,0 +1,348 @@
+package file
+
+import (
+	"crypto/rand"
+	"crypto/sha512"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/mitchellh/mapstructure"
+	flag "github.com/spf13/pflag"
+	bip39 "github.com/tyler-smith/go-bip39"
+	"golang.org/x/crypto/pbkdf2"
+
+	"github.com/oasisprotocol/deoxysii"
+	"github.com/oasisprotocol/oasis-core/go/common/crypto/sakg"
+	coreSignature "github.com/oasisprotocol/oasis-core/go/common/crypto/signature"
+
+	"github.com/oasisprotocol/oasis-sdk/cli/config"
+	"github.com/oasisprotocol/oasis-sdk/cli/wallet"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/crypto/signature"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/crypto/signature/ed25519"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/types"
+)
+
+const (
+	walletKind = "file"
+
+	cfgAlgorithm = "algorithm"
+	cfgNumber    = "number"
+
+	algorithmEd25519    = "ed25519"
+	algorithmEd25519Raw = "ed25519-raw"
+
+	stateKeySize  = 32
+	stateSaltSize = 32
+)
+
+type walletConfig struct {
+	Algorithm string `mapstructure:"algorithm"`
+	Number    uint32 `mapstructure:"number,omitempty"`
+}
+
+type secretState struct {
+	// Algorithm is the cryptographic algorithm used by the wallet. Note that this may differ from
+	// the algorithm set in the wallet's configuration in case the wallet was imported.
+	Algorithm string `json:"algorithm"`
+
+	// Data is the secret data used to derive the private key.
+	Data string `json:"data"`
+}
+
+func (s *secretState) Seal(passphrase string) (*secretStateEnvelope, error) {
+	var salt [stateSaltSize]byte
+	_, err := rand.Read(salt[:])
+	if err != nil {
+		return nil, err
+	}
+
+	envelope := &secretStateEnvelope{
+		Algorithm:  "pbkdf2",
```

Comment:
if anyone wants to bikeshed, tell us your favorite alternative

Reply:
Argon2